tmail-backend/webadmin/webadmin-email-address-contact/src/test/java/com/linagora/tmail/webadmin/EmailAddressContactRoutesTest.java

```diff
@@ -0,0 +1,865 @@
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.apache.james.core.Domain;
+import org.apache.james.core.MailAddress;
+import org.apache.james.dnsservice.api.DNSService;
+import org.apache.james.domainlist.lib.DomainListConfiguration;
+import org.apache.james.domainlist.memory.MemoryDomainList;
+import org.apache.james.webadmin.WebAdminServer;
+import org.apache.james.webadmin.WebAdminUtils;
+import org.apache.james.webadmin.utils.JsonTransformer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import com.linagora.tmail.james.jmap.contact.ContactFields;
+import com.linagora.tmail.james.jmap.contact.EmailAddressContact;
+import com.linagora.tmail.james.jmap.contact.InMemoryEmailAddressContactSearchEngine;
+
+import io.restassured.RestAssured;
+import net.javacrumbs.jsonunit.core.internal.Options;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+class EmailAddressContactRoutesTest {
+    private static final String BASE_PATH = "/domains/%s/contacts";
+    private static final String ALL_DOMAINS_PATH = "/domains/contacts";
+    private static final Domain CONTACT_DOMAIN = Domain.of("contact.com");
+
+    private static final String mailAddressA = "john@" + CONTACT_DOMAIN.asString();
+    private static final String firstnameA = "John";
+    private static final String surnameA = "Carpenter";
+
+    private static final String mailAddressB = "marie@" + CONTACT_DOMAIN.asString();
+    private static final String firstnameB = "Marie";
+    private static final String surnameB = "Carpenter";
+
+    private static Stream<Arguments> domainInvalidSource() {
+        return Stream.of(
+            Arguments.of("Dom@in"),
+            Arguments.of("@")
+        );
+    }
+
+    private WebAdminServer webAdminServer;
+    private InMemoryEmailAddressContactSearchEngine emailAddressContactSearchEngine;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        DNSService dnsService = mock(DNSService.class);
+        MemoryDomainList domainList = new MemoryDomainList(dnsService);
+        domainList.configure(DomainListConfiguration.DEFAULT);
+        domainList.addDomain(CONTACT_DOMAIN);
+
+        emailAddressContactSearchEngine = new InMemoryEmailAddressContactSearchEngine();
+
+        EmailAddressContactRoutes routes = new EmailAddressContactRoutes(emailAddressContactSearchEngine, domainList, new JsonTransformer());
+        webAdminServer = WebAdminUtils.createWebAdminServer(routes).start();
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer)
+            .setBasePath(String.format(BASE_PATH, CONTACT_DOMAIN.asString()))
```

Comment:
Please avoid using those production constants in tests and write a string `"/path/to/a"` 

 -> More explicit
 -> Test will break if path is changed. Which we want, as people will realize this is a blocking change.

Reply:
That's a local constant to the test, it's not the one declared in prod code. But I get it's confusing, will fix!